fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/AbstractLoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -204,12 +204,36 @@ public ChangedTransactionDetail reprocessLoanTransactions(final LocalDate disbur
                 reprocessChargebackTransactionRelation(changedTransactionDetail, transactionsToBeProcessed);
             } else if (loanTransaction.isChargeOff()) {
                 recalculateChargeOffTransaction(changedTransactionDetail, loanTransaction, currency, installments);
+            } else if (loanTransaction.isAccrualActivity() && loanTransaction.isNotReversed()) {
+                recalculateAccrualActivityTransaction(changedTransactionDetail, loanTransaction, currency, installments);
             }
         }
         reprocessInstallments(disbursementDate, transactionsToBeProcessed, installments, currency);
         return changedTransactionDetail;
     }
 
+    private void recalculateAccrualActivityTransaction(ChangedTransactionDetail changedTransactionDetail, LoanTransaction loanTransaction,
```

Comment:
This is okay for all the transaction processor but the `AdvancedPaymentScheduleTransactionProcessor`, there it has its own `reprocessLoanTransactions` logic, so you need to add the `Accrual Activity` explicitly.

Probably you can reuse `recalculateAccrualActivityTransaction` but maybe you need to change the visibility to be protected instead of private.